gmso/external/convert_foyer.py

```diff
@@ -0,0 +1,464 @@
+from warnings import warn
+
+import numpy as np
+import unyt as u
+import os
+import sympy
+
+from lxml import etree
+from gmso.utils.io import has_foyer  # Need for foyer
+from gmso.exceptions import ForceFieldParseError
+
+if has_foyer:
+    import foyer
+
+
+def from_foyer(foyer_xml, gmso_xml=None):
+    """Convert a foyer XML to a gmso XML
+
+    Parameters
+    ----------
+    foyer_xml : XML
+        An XML file in the foyer format 
+   
+    Returns
+    -------
+    gmso_xml : XML, default=None
+        An XML file in the gmso format
+    """
+
+    if gmso_xml is None:
+        gmso_xml = os.path.splitext(foyer_xml)[0] + "_gmso.xml"
```

Comment:
what I would recommend with Pathlib

See [this section](https://docs.python.org/3/library/pathlib.html#pathlib.PurePath.stem)
This could also be useful: https://docs.python.org/3/library/pathlib.html#pathlib.PurePath.with_suffix

I think the second link could convert what I have below into a bit less verbose code.
```python3
import pathlib



if gmso_xml is None:
    stem_file_name = pathlib.Path(str(foyer_xml)).stem
    suffix = "_gmso.xml"
    gmso_xml = pathlib.Path(stem_file_name + suffix)
else:
    gmso_xml = pathlib.Path(gmso_xml)
```

A lot of utilities can accept a path object for a filepath, but im not sure if it works for lxml.

If not, to get the actual full path to the foyer xml

```python3
filepath_as_string = str(pathlib.Path(foyer_xml))
```